File: Chapter 03/JS/section_3_3_3_lsp/before/player.js
/**
 * THE BASE CONTRACT.
 * * ARCHITECTURE NOTE: This class establishes a baseline expectation: 
 * "Any object that identifies as a Player MUST be able to execute 
 * playFieldPosition() successfully." 
 */
class Player {
    playFieldPosition() {
        // 🚨 ARCHITECTURE WARNING: We are establishing a requirement 
        // that all subclasses are expected to fulfill.
        throw new Error("playFieldPosition() must be implemented by subclass.");
    }
}

module.exports = Player;

File: Chapter 03/Python/section_3_3_3_lsp/before/player.py
from abc import ABC, abstractmethod

"""
THE BASE CONTRACT.
* ARCHITECTURE NOTE: This Abstract Base Class establishes a baseline 
expectation: "Any object that identifies as a Player MUST be able to 
execute play_field_position() successfully." 
"""
class Player(ABC):
    @abstractmethod
    def play_field_position(self):
        """
        🚨 ARCHITECTURE WARNING: We are establishing a requirement 
        that all subclasses are expected to fulfill.
        """
        pass